.github/workflows/test.yml

```diff
@@ -14,12 +14,23 @@ jobs:
       matrix:
         os: [ ubuntu-latest, macOS-latest, windows-latest ]
         python-version: [ '3.10' ]
+        pandas: ['pandas']
         include:
           - os: ubuntu-latest
             python-version: '3.8'
+            pandas: 'pandas'
             tasks: tests
           - os: ubuntu-latest
             python-version: '3.9'
+            pandas: 'pandas'
+            tasks: tests
+          - os: ubuntu-latest
+            python-version: '3.10'
+            pandas: 'pandas==2.0.3'
```

Comment:
Here I'm not completely sure, if this is the best choice.
We state as the minimum requirements 2.0.1, so maybe I should test for that version?
But on the other hand, when testing for lower versions, I usually go for the latest release of that particular version, and for 2.0.x, the latest release is 2.0.3.

Reply:
Not sure whether I understand: why do you not do `pandas>=2.0.1` ? With a fresh install it would take the newest version, in this case the latest 2.0.3 release?

I also wonder whether one could right now start removing  python3.8?  It is only getting security patches right now and will be the next to be deprecated (within the next half year).

Reply:
The test should test not the newest `pandas` version, which is 2.2.2, and which is indeed automatically covered by the tests that just install `pandas`. The idea is to make sure that `audb` also works with older `pandas` versions, e.g. 2.0.x and 2.1.x.

Reply:
Before #372, it was always working with older versions of `pandas`, but this is no longer the case. So I thought, it would be good to specify the minimum `pandas` requirements, and add tests for different `pandas` versions.

Reply:
> I also wonder whether one could right now start removing python3.8? It is only getting security patches right now and will be the next to be deprecated (within the next half year).

If there is no functional reason to remove support for a given Python version, I usually wait until the version reaches end of life, which will be [2024-10](https://devguide.python.org/versions/) for Python 3.8.

Reply:
I see now, apologies.  But then: would it not be simpler to write the matrix up using lists, similar to this:

```yaml

      - os: ubuntu-latest
        python-version: ["3.8", "3.9"]
        pandas: 'pandas'
      - os: ubuntu-latest
        python-version: "3.10"
        pandas: ["pandas==2.1.4", "pandas==2.0.3]

```

>> If there is no functional reason to remove support for a given Python version, I usually wait until the version reaches end of life, which will be [2024-10](https://devguide.python.org/versions/) for Python 3.8.

Yeah right, but then one might be asked why we do not include `3.11` that is to date the most recent version not bugfixed.

Reply:
Good, point. Python 3.11 was blocked for a long time by https://github.com/devopshq/artifactory/issues/432.

But now we have a fix for it, and I created https://github.com/audeering/audbackend/pull/227 as a first step.

Reply:
> `pandas: ["pandas==2.1.4", "pandas==2.0.3]`

One cannot use an array inside the `include:` part. It passes on the whole array, instead of creating two CI jobs, see https://github.com/audeering/audb/actions/runs/9062110425/job/24895261793.

I reverted the change to the original proposal.

Reply:
Ah alright, sorry for sending you down the wrong avenue. I have seen something like this [here](https://github.com/nogibjj/760miniP4_minlingz) and [here](https://dnsmichi.at/2021/09/23/efficient-pipelines-gitlab-ci-cd-parallel-matrix-builds-reference/) but I wouldn't know whether this works in the current scenario. 

The general review will approve.

Reply:
Yes, when defining the matrix, we also use lists:

```yaml
    strategy:
      matrix:
        os: [ ubuntu-latest, macOS-latest, windows-latest ]
        python-version: [ '3.10' ]
        pandas: ['pandas']
```

But I excluded the other Python versions, and the other pandas versions from being tested on all platforms, as the tests for Windows and MacOS can be quite slow. It's not perfect to not test everything, but I think it is an ok compromise to not have to always wait for the CI pipelines.